GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -763,6 +785,26 @@ private FileSystemTaskResult ExecuteBackgroundOperation(FileSystemTask gitUpdate
             return result;
         }
 
+        private FileSystemTaskResult TryRemoveModifiedPath(string virtualPath, bool isFolder)
+        {
+            string fullPathToItem = Path.Combine(this.context.Enlistment.WorkingDirectoryRoot, virtualPath);
+            if ((isFolder && this.context.FileSystem.DirectoryExists(fullPathToItem)) || 
+                (!isFolder && this.context.FileSystem.FileExists(fullPathToItem)))
+            {
+                return FileSystemTaskResult.Success;
+            }
+
+            if (!this.modifiedPaths.TryRemove(virtualPath, isFolder, out bool isRetryable))
+            {
+                return isRetryable ? FileSystemTaskResult.RetryableError : FileSystemTaskResult.FatalError;
+            }
+
+            this.newlyCreatedFileAndFolderPaths.TryRemove(virtualPath);
+
+            this.InvalidateGitStatusCache();

```

Comment:
Why do we have to invalidate the status cache now (or, why didn't we have to before)?

Reply:
Anything that would affect the status cache should invalidate it.  In this case we are removing paths from the modified paths.  So this is more of being paranoid because with the way it is coded right now the `newlyCreatedFileAndFolderPaths` would get cleared with any status update and this code path not hit.

But this brings up a question for @jamill, if the background status runs will that clear the `newlyCreatedFileAndFolderPaths`, which right now if it changes the index it would, so this makes the argument to only clear when the projection changes stronger.  If that change is made then the invalidation of the status cache would be required because a user could create a file, run git status, delete file, run git status and since the modified paths changed with the delete file the status cache would need to be invalidated.  Right now when that delete happens we add to the modified paths which invalidates the status cache.

I will do some testing around this to make sure the modified paths are getting clean up even with the background status running.

Reply:
> or, why didn't we have to before

I found that we're currently invalidating the status cache in TryAddModifiedPath, and @kewillford your explanation about why we need it for delete as well makes sense. Thanks!